app/src/marketplace/containers/ProductPage/MembersGraphV2.jsx

```diff
@@ -18,25 +18,55 @@ const MembersGraphV2 = ({ dataUnionAddress, memberCount, shownDays = 7 }: Props)
     const [memberCountUpdatedAt, setMemberCountUpdatedAt] = useState(Date.now())
     const [memberData, setMemberData] = useState([])
     const [graphData, setGraphData] = useState([])
+    const generator = useRef(null)
 
     const startDate = useMemo(() => (
         Date.now() - (shownDays * MILLISECONDS_IN_DAY)
     ), [shownDays])
 
+    const reset = useCallback(() => {
+        setGraphData([])
+        setMemberData([])
+    }, [])
+
     useEffect(() => {
         setMemberCountUpdatedAt(Date.now())
     }, [memberCount])
 
     useEffect(() => {
         const loadData = async () => {
-            const result = await getJoinsAndParts(dataUnionAddress, startDate)
-            setMemberData(result)
+            try {
+                if (generator.current != null) {
+                    generator.current.return('Canceled')
+                    generator.current = null
+                    reset()
+                }
+                generator.current = getJoinsAndParts(dataUnionAddress, startDate)
+
+                // eslint-disable-next-line no-restricted-syntax
+                for await (const event of generator.current) {
+                    setMemberData((prev) => [
```

Comment:
Each round of that `for` seem to be asynchronous. We may have to check if we're mounted here.

Reply:
True, good point!